### PR TITLE
Bgd 4094 deploy sidecars on all data plane components

### PR DESCRIPTION
**Pull request by Z4ck404**
- [x] Add custom configs to all the telemetry sidecars in order to enable sidecars on all the components and avoid logs and metrics redundancy. 
- [x] Add metrics collection pipeline config

# Jira Ticket

https://spotinst.atlassian.net/browse/BGD-4094

### CODE REVIEW

File: charts/bigdata-telemetry/templates/thanos-receiver-configmap.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ data:
     [
       {
         "endpoints": [
-          "{{ include "bigdata-telemetry.fullname" . }}-thanos-receiver-svc.{{ .Release.Namespace }}.svc.cluster.local:10901"
+          "127.0.0.1:10901"
```
Comment (ImpSy) [inline charts/bigdata-telemetry/templates/thanos-receiver-configmap.yaml:13]: why this change ?
And why not `localhost:10901` ?

File: charts/bigdata-notebook-service-storage-server/templates/custom_telemetry_configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{{- if .Values.telemetry.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "bigdata-notebook-service-storage-server.fullname" . }}-custom-fluentbit-cm
```
Comment (ImpSy) [inline charts/bigdata-notebook-service-storage-server/templates/custom_telemetry_configmap.yaml:5]: ## ⚠️ This comment is for all services ⚠️ 
Why name them `<components>-custom-fluentbit-cm` and not `<components>-fluentbit-cm` (there are the config map of the service, no need to add custom)

This also apply to the file name that could be `telemetry-configmap`

Finally why not name the configmap `<components>-telemetry-cm` to keep things more clear

File: charts/bigdata-notebook-service/templates/custom_telemetry_configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  {{- include "bigdata-notebook-service.labels" . | nindent 4 }}
+data:
+  custom-filters.conf: |
+    [FILTER]
+        Name    grep
+        Match   kube.*
+        Regex   $kubernetes['labels']['bigdata.spot.io/component'] {{ include "bigdata-notebook-service.fullname" . }}
+
+  metrics-collection.conf: |
+    # Configuration for collecting metrics from the notebook service
```
Comment (ImpSy) [inline charts/bigdata-notebook-service/templates/custom_telemetry_configmap.yaml:16]: why keep an empty section here ?

File: charts/bigdata-notebook-service/templates/deployment.yaml [hunk 4]
```diff
@@ -171,6 +171,12 @@ spec:
             - name: config
               mountPath: /fluent-bit/etc/fluent-bit.conf
               subPath: fluent-bit.conf
+            - name: custom-config
+              mountPath: /fluent-bit/etc/custom-filters.conf
+              subPath: custom-filters.conf
+            - name: custom-config
+              mountPath: /fluent-bit/etc/metrics-collection.conf
+              subPath: metrics-collection.conf
```
Comment (ImpSy) [inline charts/bigdata-notebook-service/templates/deployment.yaml:179]: why mount an empty section ?

File: charts/bigdata-notebook-service-storage-server/templates/notebook-service-storage-deployment.yaml [hunk 5]
```diff
@@ -81,22 +71,36 @@ spec:
...
           - name: varlibdockercontainers
             readOnly: true
             mountPath: /var/lib/docker/containers
+          - name: telemetry-aws-credentials
+            mountPath: /root/.aws
       volumes:
         - name: config
           configMap:
             name: bigdata-telemetry-fluentbit-cm
+        - name: custom-config
```
Comment (ImpSy) [inline charts/bigdata-notebook-service-storage-server/templates/notebook-service-storage-deployment.yaml:92]: this should have better name like `telementry-global-config` and `telemetry-service-config`

File: charts/bigdata-operator/templates/common_telemetry_configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+{{- if .Values.telemetry.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: bigdata-telemetry-fluentbit-cm
```
Comment (ImpSy) [inline charts/bigdata-operator/templates/common_telemetry_configmap.yaml:5]: you need to make the name of the file match the name of the configmap
Is it a common telemetry configmap ? or a common fluentbit configmap for telemetry ?
The choice is then:
- `common_telemetry_configmap.yaml` 
`bigdata-common-telemetry-cm` ou `bigdata-telemetry-cm`
- `common_telemetry_fluentbit_configmap.yaml` 
`bigdata-common-telemetry-fluentbit-cm` ou `bigdata-telemetry-fluentbit-cm`

File: charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      host                    0.0.0.0
+      port                    10254
+      tag                     {{ include "spark-operator.fullname" . }}
+      metrics_path            /metrics
+      scrape_interval         5s
+
+    [OUTPUT]
+      Name                 prometheus_remote_write
+      Match                {{ include "spark-operator.fullname" . }}
+      Host                 bigdata-telemetry-thanos-receiver-svc.spot-system.svc.cluster.local
```
Comment (ImpSy) [inline charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml:27]: do not hardcode namespace

File: charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    [FILTER]
+        Name    grep
+        Match   kube.*
+        Regex   $kubernetes['labels']['bigdata.spot.io/component'] {{ include "spark-operator.fullname" . }}
+
+  metrics-collection.conf: |
+    [INPUT]
+      name                    prometheus_scrape
+      host                    0.0.0.0
+      port                    10254
```
Comment (ImpSy) [inline charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml:19]: retrieve this from the values.yaml

File: charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+      tag                     {{ include "spark-operator.fullname" . }}
+      metrics_path            /metrics
```
Comment (ImpSy) [inline charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml:21]: retrieve this from values.yaml

File: charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+metadata:
+  name: {{ include "spark-operator.fullname" . }}-custom-fluentbit-cm
+  labels:
+  {{- include "spark-operator.labels" . | nindent 4 }}
+data:
+  custom-filters.conf: |
+    [FILTER]
+        Name    grep
+        Match   kube.*
+        Regex   $kubernetes['labels']['bigdata.spot.io/component'] {{ include "spark-operator.fullname" . }}
```
Comment (ImpSy) [inline charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml:13]: retrieve this from values.yaml

File: charts/bigdata-notebook-service-storage-server/templates/custom_telemetry_configmap.yaml (hunk 2 above)
Comment (Z4ck404) [reply charts/bigdata-notebook-service-storage-server/templates/custom_telemetry_configmap.yaml:5]: Because in the operator, I create two CM.
one common one and the custom one containing specific filters for the operator (logs / metrics). so I applied the same pattern for the rest of the components.

File: charts/bigdata-notebook-service/templates/custom_telemetry_configmap.yaml (hunk 3 above)
Comment (Z4ck404) [reply charts/bigdata-notebook-service/templates/custom_telemetry_configmap.yaml:16]: Because I do reference this section in the common CM  (I create once in the bigdata-operator charts).
if I doesn't exist the sidecar fails. i could mount an empty folders and name it  `metrics-collection.conf` but I thought why not have the section correctly as a placeholder

File: charts/bigdata-notebook-service/templates/deployment.yaml (hunk 4 above)
Comment (Z4ck404) [reply charts/bigdata-notebook-service/templates/deployment.yaml:179]: the reason here https://github.com/spotinst/bigdata-charts/pull/190#discussion_r1544651225

File: charts/bigdata-telemetry/templates/thanos-receiver-configmap.yaml (hunk 1 above)
Comment (Z4ck404) [reply charts/bigdata-telemetry/templates/thanos-receiver-configmap.yaml:13]: Both 127.0.0.1 and localhost are correct here, but I am  using the IP address can be more explicit and avoids any potential DNS resolution issues.

File: charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml (hunk 10 above)
Comment (ImpSy) [reply charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml]: why not

Comment (Z4ck404) [reply charts/spark-operator/charts/spark-operator/templates/custom_telemetry_configmap.yaml]: you still have the the label key  there harcoded. I don't see the point TBH.
we will make sure to update all the charts when this label is updated !

File: charts/bigdata-operator/Chart.yaml [hunk 11]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: bigdata-operator
 description: Spot Ocean BigData Operator
 type: application
-version: 0.4.10
-appVersion: 0.4.9
+version: 0.4.11
+appVersion: 0.4.7
```
Comment (alextarasov-spot) [inline charts/bigdata-operator/Chart.yaml:6]: the appVersion version is downgraded

File: charts/bigdata-operator/values.yaml [hunk 12]
```diff
@@ -45,64 +45,11 @@ securityContext:
 # runAsUser: 1000
 
 telemetry:
-  enabled: false
+  enabled: true
```
Comment (alextarasov-spot) [inline charts/bigdata-operator/values.yaml:48]: do we want to set it to true here?

File: charts/bigdata-notebook-workspace/templates/telemetry_configmap.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  name: {{ include "bigdata-notebook-workspace.fullname" . }}-telemetry-cm
+  labels:
+  {{- include "bigdata-notebook-workspace.labels" . | nindent 4 }}
+data:
+  custom-filters.conf: |
+    [FILTER]
+        Name    grep
+        Match   kube.*
+        Regex   $kubernetes['labels']['bigdata.spot.io/component'] {{ index .Values.podLabels "bigdata.spot.io/component" | quote }}
```
Comment (ImpSy) [inline charts/bigdata-notebook-workspace/templates/telemetry_configmap.yaml:13]: add the pod label in values.yaml (only service missing this annotation)

**Review by ImpSy — APPROVED** [review]

File: charts/bigdata-operator/values.yaml (hunk 12 above)
Comment (Z4ck404) [reply charts/bigdata-operator/values.yaml:48]: this was left during testing .. will fix it

File: charts/bigdata-notebook-workspace/templates/telemetry_configmap.yaml (hunk 13 above)
Comment (Z4ck404) [reply charts/bigdata-notebook-workspace/templates/telemetry_configmap.yaml:13]: Will add it

File: charts/bigdata-operator/Chart.yaml [hunk 14]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: bigdata-operator
 description: Spot Ocean BigData Operator
 type: application
-version: 0.4.10
+version: 0.4.11
 appVersion: 0.4.9
```
Comment (alextarasov-spot) [inline charts/bigdata-operator/Chart.yaml:6]: nit: the current version is 0.4.10
 https://github.com/spotinst/bigdata-operator/blob/0c4f6b5d6366ba1195e7bb91255da0f11a7c9fae/VERSION_NUMBER#L1

**Review by alextarasov-spot — APPROVED** [review]

**Review by ImpSy — APPROVED** [review]
